src/message_interface/message_handler.rs

```diff
@@ -262,6 +263,15 @@ impl WalletMessageHandler {
                 })
                 .await
             }
+            Message::Bech32ToHex(bech32) => {
+                convert_panics(|| Ok(Response::Bech32ToHex(utils::bech32_to_hex(&bech32)?)))
+            }
+            Message::HexToBech32 { hex, bech32_hrp } => convert_panics(|| {
+                Ok(Response::HexToBech32(utils::hex_to_bech32(
+                    &hex,
+                    &bech32_hrp.unwrap_or_else(|| SHIMMER_TESTNET_BECH32_HRP.into()),
```

Comment:
Instead of always just using `SHIMMER_TESTNET_BECH32_HRP` I think this should call `get_node_info()` from the account manager and use the bech32 hrp from there, only if the request fails use `SHIMMER_TESTNET_BECH32_HRP`